hw25 finished

nkrenta opened the pull request:
solved problems:
1. Cycled output of findByAgeBetween
2. Rebuild getStudentFaculty to get the correct output (rewrite faculty.toString)
added:
sub method to output faculty's name by student id

nkrenta committed:
hw25 finish

Antony8720 commented on src/main/java/ru/hogwarts/school/model/Student.java:
```diff
@@ -14,7 +11,12 @@ public class Student {
     @GeneratedValue(strategy = GenerationType.AUTO)
     private long id;
 
+    @ManyToOne
```
Нужно использовать lazy fetch type

Antony8720 commented on src/main/java/ru/hogwarts/school/controller/StudentController.java:
```diff
@@ -50,6 +52,27 @@ public ResponseEntity<List<Student>> findByAge(@RequestParam(required = false) I
         return ResponseEntity.ok(Collections.emptyList());
     }
 
+    @GetMapping("/{min}/{max}")
```
Данные параметры стоит передавать через `@RequestParam`

Antony8720 commented on src/main/java/ru/hogwarts/school/scripts.sql:
```diff
@@ -0,0 +1,22 @@
+
+select s.name
+from student as s;
+
```
Данный файл нужно расположить в папке resources или в корне проекта

Antony8720 commented on src/main/java/ru/hogwarts/school/controller/FacultyController.java:
```diff
@@ -42,12 +43,17 @@ public ResponseEntity<Collection<Faculty>> getAllFacultiesInfo() {
-        }
-        return ResponseEntity.ok(Collections.emptyList());
+    @GetMapping()
+    public ResponseEntity<Collection<Faculty>> findByNameOrColorIgnoreCase(@RequestParam(required = false) String name, @RequestParam(required = false) String color) {
```
Если поглубже изучить аннотацию `@RestController`, то она оборачивает (с помощью аннотации `@ResponseBody`) тело каждого метода в ResponseEntity. Таким образом нам нет смысла явно использовать этот класс в сигнатуре метода, и избавившись от него, наш код станет более удобочитаемым

nkrenta committed:
hw25 not finished
troubles with output:
findByAgeBetween
findByNameOrColorIgnoreCase

Output one of them
[
    {
        "id": 1,
        "faculty": {
            "id": 4,
            "students": [
                {
                    "id": 1,
                    "faculty": {
                        "id": 4,
                        "students": [
                            {
                                "id": 1,
                                "faculty": {
                                    "id": 4,
                                    "students": [
                                        {
                                            "id": 1,
                                            "faculty": {
                                                "id": 4,
                                                "students": [
                                                    {

nkrenta committed:
hw25 finished.
solved problems:
1. Cycled output of findByAgeBetween
2. Rebuild getStudentFaculty to get the correct output (rewrite faculty.toString)
added:
sub method to output faculty's name by student id
